### PR TITLE
Moved BL from the drawer's view-model to interactors.

### PR DESCRIPTION
**Pull request by BarAgam**
Moved BL from the drawer's view-model to interactors.

### CODE REVIEW

File: app/src/test/java/com/teamagam/gimelgimel/app/integration/interactors/DisplayVectorLayerInteractorTest.java [hunk 1]
```diff
@@ -140,9 +140,11 @@ private VectorLayer createVectorLayer(int num) {
   }
 
   private void executeSetVectorLayerVisibilityInteractor(String id, boolean isVisible) {
```
Comment (teamagam5) [inline app/src/test/java/com/teamagam/gimelgimel/app/integration/interactors/DisplayVectorLayerInteractorTest.java]: rename method respectively?

File: domain/src/main/java/com/teamagam/gimelgimel/domain/layers/OnVectorLayerListingClickInteractor.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      VectorLayerPresentation vectorLayerPresentation) {
+    super(threadExecutor);
+    mVectorLayersVisibilityRepository = vectorLayersVisibilityRepository;
+    mVectorLayerPresentation = vectorLayerPresentation;
+  }
+
+  @Override
+  protected Iterable<SubscriptionRequest> buildSubscriptionRequests(DataSubscriptionRequest.SubscriptionRequestFactory factory) {
+    SubscriptionRequest setVisibilityRequest =
+        factory.create(Observable.just(mVectorLayerPresentation),
```
Comment (teamagam5) [inline domain/src/main/java/com/teamagam/gimelgimel/domain/layers/OnVectorLayerListingClickInteractor.java:31]: When rebasing on Rx2, I think maybe you should use the new SIGNAL convention. But only if it doesn't reduce readability.

File: domain/src/main/java/com/teamagam/gimelgimel/domain/layers/OnVectorLayerListingClickInteractor.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            vectorLayerVisibilityChangeObservable -> vectorLayerVisibilityChangeObservable.doOnNext(
+                this::toggleVisibility));
```
Comment (teamagam5) [inline domain/src/main/java/com/teamagam/gimelgimel/domain/layers/OnVectorLayerListingClickInteractor.java:33]: I guess the GoTo behavior will be implemented in a future task?

File: domain/src/main/java/com/teamagam/gimelgimel/domain/rasters/OnRasterListingClickedInteractor.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      IntermediateRasterPresentation intermediateRasterPresentation) {
+    super(threadExecutor);
+    mVisibilityRepository = visibilityRepository;
+    mIntermediateRasterPresentation = intermediateRasterPresentation;
+  }
+
+  @Override
+  protected Iterable<SubscriptionRequest> buildSubscriptionRequests(DataSubscriptionRequest.SubscriptionRequestFactory factory) {
+    SubscriptionRequest setRasterRequest =
+        factory.create(Observable.just(mIntermediateRasterPresentation),
```
Comment (teamagam5) [inline domain/src/main/java/com/teamagam/gimelgimel/domain/rasters/OnRasterListingClickedInteractor.java:30]: ditto

File: domain/src/main/java/com/teamagam/gimelgimel/domain/user/OnUserListingClickedInteractor.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      @Provided com.teamagam.gimelgimel.domain.map.GoToLocationMapInteractorFactory goToLocationMapInteractorFactory,
+      UserLocation userLocation) {
+    super(threadExecutor);
+    mUserLocation = userLocation;
+    mGoToLocationMapInteractorFactory = goToLocationMapInteractorFactory;
+  }
+
+  @Override
+  protected Iterable<SubscriptionRequest> buildSubscriptionRequests(DataSubscriptionRequest.SubscriptionRequestFactory factory) {
+    return Arrays.asList(factory.create(Observable.just(mUserLocation),
```
Comment (teamagam5) [inline domain/src/main/java/com/teamagam/gimelgimel/domain/user/OnUserListingClickedInteractor.java:29]: `Collections.singletonList(...)`?

**Review by teamagam5 — APPROVED** [review]

File: app/src/test/java/com/teamagam/gimelgimel/app/integration/interactors/DisplayVectorLayerInteractorTest.java (hunk 1 above)
Comment (BarAgam) [reply app/src/test/java/com/teamagam/gimelgimel/app/integration/interactors/DisplayVectorLayerInteractorTest.java]: Correctomundo

File: domain/src/main/java/com/teamagam/gimelgimel/domain/layers/OnVectorLayerListingClickInteractor.java (hunk 3 above)
Comment (BarAgam) [reply domain/src/main/java/com/teamagam/gimelgimel/domain/layers/OnVectorLayerListingClickInteractor.java:33]: Yeap

**Pull request merged**